### PR TITLE
Update charmrepo dependency.

### PR DESCRIPTION
**Pull request by SimonRichardson**
The charmrepo dependency brought in a init function that added http
`/debug/pprof/`. This, unfortunately, brought it in twice, which is a bit
problematic as it caused a panic.

## QA steps

Can bootstrap and deploy a `cs:` charm.

```sh
$ juju bootstrap lxd test
$ juju deploy cs:ubuntu
```

## Bug reference

https://bugs.launchpad.net/juju/+bug/1944740

### CODE REVIEW

**Review by manadart — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -47,24 +75,32 @@ require (
...
 	github.com/im7mortal/kmutex v1.0.1
 	github.com/imdario/mergo v0.3.10 // indirect
+	github.com/jessevdk/go-flags v1.4.0 // indirect
+	github.com/jmespath/go-jmespath v0.4.0 // indirect
+	github.com/json-iterator/go v1.1.10 // indirect
 	github.com/juju/ansiterm v0.0.0-20210706145210-9283cdf370b5
 	github.com/juju/blobstore/v2 v2.0.0-20210302045357-edd2b24570b7
 	github.com/juju/charm/v9 v9.0.0-20210922083844-6bd7c961dbc6
-	github.com/juju/charmrepo/v7 v7.0.0-20210901102801-513fb9274dae
+	github.com/juju/charmrepo/v7 v7.0.0-20210923152136-0ae2f26643bf
```
Comment (SimonRichardson) [inline go.mod:84]: The real fix, but I did a driveby and fixed the whole file.

**Comment by SimonRichardson** [comment]
$$merge$$

File: go.mod [hunk 2]
```diff
@@ -83,17 +119,37 @@ require (
 	github.com/juju/schema v1.0.1-0.20190814234152-1f8aaeef0989
 	github.com/juju/terms-client/v2 v2.0.0-20210422053140-27f71c100676
 	github.com/juju/testing v0.0.0-20210324180055-18c50b0c2098
+	github.com/juju/txn v0.0.0-20210302043154-251cea9e140a // indirect
 	github.com/juju/txn/v2 v2.0.0-20210407000251-11166e89894c
```
Comment (jameinel) [inline go.mod:123]: this sort of thing concerns me, that we would depend on multiple versions of the same library, and that seems to happen a lot.

**Review by jameinel — COMMENTED** [review]
I'm happy to avoid the double stanza, I feel like we still don't have something correct given the number of double imports.

**Comment by hpidcock** [comment]
The double require stanza is introduced by go1.17. I suggest we leave it.

File: go.mod (hunk 2 above)
Comment (SimonRichardson) [reply go.mod]: `go mod why` is our friend here:

```
go mod why github.com/juju/txn
# github.com/juju/txn
github.com/juju/juju/scripts/juju-blobstore-cleanup
github.com/juju/blobstore/v2
github.com/juju/txn
```

Comment (SimonRichardson) [reply go.mod]: [This is still using txn v1](https://github.com/juju/blobstore/blob/v2/managedstorage.go#L24), probably needs updating tbh

File: go.mod [hunk 3]
```diff
@@ -47,24 +75,32 @@ require (
...
+	github.com/juju/charmrepo/v7 v7.0.0-20210923152136-0ae2f26643bf
 	github.com/juju/clock v0.0.0-20190205081909-9c5c9712527c
+	github.com/juju/cmd v0.0.0-20200108104440-8e43f3faa5c9 // indirect
 	github.com/juju/cmd/v3 v3.0.0-20210809234809-65029dab4cd0
 	github.com/juju/collections v0.0.0-20200605021417-0d0ec82b7271
 	github.com/juju/description/v3 v3.0.0-20210709162012-5f861fa82eab
 	github.com/juju/errors v0.0.0-20210818161939-5560c4c073ff
 	github.com/juju/featureflag v0.0.0-20200423045028-e2f9e1cb1611
 	github.com/juju/gnuflag v0.0.0-20171113085948-2ce1bb71843d
+	github.com/juju/go4 v0.0.0-20160222163258-40d72ab9641a // indirect
```
[the remark this answers is not in the document]
Comment (SimonRichardson) [reply go.mod]: So we've ended up with a fork that hasn't been updated in 6 years, even though upstream has new content. :disappointed:

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$
